mailbag/formats/pst.py

```diff
@@ -27,81 +28,131 @@ def __init__(self, target_account, args, **kwargs):
+                        try:
+                            attachmentNames = []
+                            attachments = []
+                            total_attachment_size_bytes = 0
+                            for attachment in messageObj.attachments:
+                                total_attachment_size_bytes = total_attachment_size_bytes + attachment.get_size()
+                                attachment_content = attachment.read_buffer(attachment.get_size())
+                                attachments.append(attachment_content)
+                                attachmentNames.append(attachment.get_name())
+                        except Exception as e:
+                            log.error(e)
+                            error.append("Error parsing attachments.")
 
                         message = Email(
-                            Message_ID=headers['Message-ID'],
-                            Email_Folder=join(*path),
+                            Error=error,
+                            Message_ID=headers['Message-ID'].strip(),
+                            Message_Path=os.path.join(os.path.splitext(original_filename)[0], *path),
+                            Original_Filename=original_filename,
                             Date=headers["Date"],
                             From=headers["From"],
                             To=headers["To"],
                             Cc=headers["To"],
                             Bcc=headers["Bcc"],
                             Subject=headers["Subject"],
-                            Content_Type=headers["Content-Type"],
+                            Content_Type=headers.get_content_type(),
                             Headers=headers,
-                            # detecting encoding might be problematic but works for now
-                            Body=str(messageObj.html_body),
-                            Text_Body=messageObj.plain_text_body.decode(chardet.detect(messageObj.plain_text_body)['encoding']),
-                            HTML_Body=messageObj.html_body.decode(chardet.detect(messageObj.html_body)['encoding']),
+                            HTML_Bytes=html_bytes,
+                            HTML_Body=html_body,
+                            Text_Bytes=text_bytes,
+                            Text_Body=text_body,
                             AttachmentNum=int(messageObj.number_of_attachments),
                             Message=None,
                             AttachmentNames=attachmentNames,
-                            AttachmentFiles=attachments,
-                            Error='False'
+                            AttachmentFiles=attachments
                         )
                     
                     except (Exception) as e:
                         log.error(e)
                         message = Email(
-                            Error='True'
+                            Error=['Error parsing message.']
```

Comment:
Should we also include errors from previous try exceptions into the model here?

Reply:
Yup, that was a bit lazy. Should be fixed now.